subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/VersionResolver.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.fixture
+
+import com.google.common.base.Splitter
+import groovy.transform.CompileStatic
+import groovy.transform.TypeCheckingMode
+import org.gradle.integtests.fixtures.executer.GradleDistribution
+import org.gradle.integtests.fixtures.versions.ReleasedVersionDistributions
+import org.gradle.util.GradleVersion
+
+import java.util.regex.Pattern
+
+class VersionResolver {
```

Comment:
`BaselineVersionResolver` would be a better name I think.

Reply:
I thought about that too - I think it's better now